examples/dcp/stateful_example.py

```diff
@@ -0,0 +1,138 @@
+# based on PyTorch sample https://github.com/pytorch/pytorch/blob/main/torch/distributed/checkpoint/examples/stateful_example.py
```

Comment:
should we add the Amazon copyright header?

Reply:
That is complicated question, as the source of that example belong to PyTorch. Let me try to find how we dealt with that in other cases.